src/stores/activitiesStores/ActivityHistoryStore.js

```diff
@@ -0,0 +1,104 @@
+import { observable, action, reaction, computed } from 'mobx';
+import { TransactionType, SortBy, Routes } from 'constants';
+import _ from 'lodash';
+
+import { getDetailPagePath } from '../../helpers/utility';
+import { queryAllTransactions, queryAllOracles } from '../../network/graphQuery';
+import Transaction from '../models/Transaction';
+import Oracle from '../models/Oracle';
+
+const INIT_VALUES = {
+  transactions: [],
+  order: SortBy.DESCENDING.toLowerCase(),
+  orderBy: 'createdTime',
+  perPage: 10,
+  page: 0,
+  limit: 50,
+  perLimit: 50,
+  skip: 0,
+};
+
+export default class {
+  @observable transactions = INIT_VALUES.transactions
+  @observable order = INIT_VALUES.order
+  @observable orderBy = INIT_VALUES.orderBy
+  @observable perPage = INIT_VALUES.perPage
+  @observable page = INIT_VALUES.page
+  @observable skip = INIT_VALUES.skip
+  @observable limit = INIT_VALUES.limit
+  @observable perLimit = INIT_VALUES.perLimit
+
+  constructor(app) {
+    this.app = app;
+    reaction( // Update page on new block
+      () => this.app.global.syncBlockNum,
+      async () => this.transactions = await this.fetchHistory()
+    );
+    reaction( // Sort while order changes
+      () => this.order + this.orderBy,
+      async () => this.transactions = await this.fetchHistory()
+    );
+    reaction( // Refresh when need more data
```

Comment:
instead of doing it this way, why don't we have this
```js
@computed get hasMore() {
  return this.transactions.length % this.perPage > 0
}
```
then do
```js
reaction(
  () => this.page + this.perPage,
  async () => {
    if (this.hasMore) {
      const newTransactions = await this.fetchHistory(this.orderBy, this.order, this.perPage, this.transactions.length)
      this.transactions = [...this.transactions, ...newTransactions]
      this.skip = this.transactions.length
    }
  }
)
```